github/private_registries.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2025 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// PrivateRegistriesService handles communication with the private registries
+// methods of the GitHub API.
+//
+// GitHub API docs: https://docs.github.com/rest/private-registries
+type PrivateRegistriesService service
+
+// PrivateRegistryType represents the type of private registry.
+type PrivateRegistryType string
+
+const (
+	PrivateRegistryTypeMavenRepository    PrivateRegistryType = "maven_repository"
+	PrivateRegistryTypeNugetFeed          PrivateRegistryType = "nuget_feed"
+	PrivateRegistryTypeGoProxyServer      PrivateRegistryType = "goproxy_server"
+	PrivateRegistryTypeNpmRegistry        PrivateRegistryType = "npm_registry"
+	PrivateRegistryTypeRubygemsServer     PrivateRegistryType = "rubygems_server"
+	PrivateRegistryTypeCargoRegistry      PrivateRegistryType = "cargo_registry"
+	PrivateRegistryTypeComposerRepository PrivateRegistryType = "composer_repository"
+	PrivateRegistryTypeDockerRegistry     PrivateRegistryType = "docker_registry"
+	PrivateRegistryTypeGitSource          PrivateRegistryType = "git_source"
+	PrivateRegistryTypeHelmRegistry       PrivateRegistryType = "helm_registry"
+	PrivateRegistryTypeHexOrganization    PrivateRegistryType = "hex_organization"
+	PrivateRegistryTypeHexRepository      PrivateRegistryType = "hex_repository"
+	PrivateRegistryTypePubRepository      PrivateRegistryType = "pub_repository"
+	PrivateRegistryTypePythonIndex        PrivateRegistryType = "python_index"
+	PrivateRegistryTypeTerraformRegistry  PrivateRegistryType = "terraform_registry"
+)
+
+// PrivateRegistryVisibility represents the visibility of a private registry.
+type PrivateRegistryVisibility string
+
+const (
+	PrivateRegistryVisibilityPrivate  PrivateRegistryVisibility = "private"
+	PrivateRegistryVisibilityAll      PrivateRegistryVisibility = "all"
+	PrivateRegistryVisibilitySelected PrivateRegistryVisibility = "selected"
+)
+
+// PrivateRegistry represents a private registry configuration.
+type PrivateRegistry struct {
+	Name         *string    `json:"name,omitempty"`
+	RegistryType *string    `json:"registry_type,omitempty"`
+	Username     *string    `json:"username,omitempty"`
+	CreatedAt    *Timestamp `json:"created_at,omitempty"`
+	UpdatedAt    *Timestamp `json:"updated_at,omitempty"`
+	Visibility   *string    `json:"visibility,omitempty"`
+}
+
+// PrivateRegistries represents a list of private registries.
+type PrivateRegistries struct {
+	TotalCount     *int               `json:"total_count,omitempty"`
+	Configurations []*PrivateRegistry `json:"configurations,omitempty"`
+}
+
+// CreateOrganizationPrivateRegistry represents the payload to create a private registry.
+type CreateOrganizationPrivateRegistry struct {
+	RegistryType string `json:"registry_type"`
```

Comment:
Could we add a comment to the field?


```suggestion
	// The registry type.
	// Can be one of: maven_repository, nuget_feed, goproxy_server, npm_registry, rubygems_server,
	// cargo_registry, composer_repository, docker_registry, git_source, helm_registry, hex_organization,
	// hex_repository, pub_repository, python_index, terraform_registry.
	RegistryType string `json:"registry_type"`
```